### PR TITLE
More comments in CUDA backends

### PR DESCRIPTION
**Pull request by jeremylt**
I added some comments to the CUDA backend to help me this summer.

### CODE REVIEW

File: backends/cuda-gen/ceed-cuda-gen.c [hunk 1]
```diff
@@ -22,11 +22,18 @@
 #include <cuda_runtime.h>
 #include "ceed-cuda-gen.h"
 
+//------------------------------------------------------------------------------
+// CUDA perferred MemType
```
Comment (YohannDudouit) [inline backends/cuda-gen/ceed-cuda-gen.c:26]: typo

File: backends/cuda-reg/ceed-cuda-reg-basis.c [hunk 2]
```diff
@@ -376,107 +452,90 @@ inline __device__ void grad3d(const CeedInt nelem, const int transpose,
...
+  } else if (BASIS_DIM == 3) {
     weight3d(nelem, qweight1d, v);
   }
 }
 
 );
 // *INDENT-ON*
 
+//------------------------------------------------------------------------------
+// Device  initalization
```
Comment (YohannDudouit) [inline backends/cuda-reg/ceed-cuda-reg-basis.c:527]: typo

File: backends/cuda/ceed-cuda-restriction.c [hunk 3]
```diff
@@ -21,6 +21,9 @@
 static const char *restrictionkernels = QUOTE(
 
 #if __CUDA_ARCH__ < 600
+//------------------------------------------------------------------------------
```
Comment (YohannDudouit) [inline backends/cuda/ceed-cuda-restriction.c:24]: Note to myself: This `#if` doesn't work, doing it like this replaces the `atomicAdd` in all cases.

File: backends/cuda-gen/ceed-cuda-gen.c (hunk 1 above)
Comment (jeremylt) [reply backends/cuda-gen/ceed-cuda-gen.c]: thanks

**Comment by jeremylt** [comment]
I verified that I haven't broken anything with these comments, and I found some minor QFunction bugs. I'm happy to merge this now if people like it.

File: backends/cuda-gen/ceed-cuda-gen-operator.c [hunk 4]
```diff
@@ -19,14 +19,21 @@
...
 
+//------------------------------------------------------------------------------
+// Destroy operator
+//------------------------------------------------------------------------------
 static int CeedOperatorDestroy_Cuda_gen(CeedOperator op) {
   int ierr;
   CeedOperator_Cuda_gen *impl;
   ierr = CeedOperatorGetData(op, (void *)&impl); CeedChk(ierr);
   ierr = CeedFree(&impl); CeedChk(ierr);
+
```
Comment (jedbrown) [inline backends/cuda-gen/ceed-cuda-gen-operator.c:30]: I don't see a purpose to this hanging whitespace.

Comment (jeremylt) [reply backends/cuda-gen/ceed-cuda-gen-operator.c:30]: No purpose beyond matching style between functions. Almost all of our functions have it, for some reason.

Comment (jedbrown) [reply backends/cuda-gen/ceed-cuda-gen-operator.c:30]: I don't know why. I usually remove it if editing code that has it.

Comment (jeremylt) [reply backends/cuda-gen/ceed-cuda-gen-operator.c:30]: We could just start letting it die from this point forward? I don't much care either way; I just try to match style as I notice it.

Comment (jedbrown) [reply backends/cuda-gen/ceed-cuda-gen-operator.c:30]: Sounds good.

**Pull request merged**